schemas/2015-04-08/Microsoft.DocumentDB.json

```diff
@@ -89,5 +69,67 @@
         "location"
       ]
     }
+  },
+  "definitions": {
+    "consistencyBoundedStaleness": {
+      "type": "object",
+      "properties": {
+        "defaultConsistencyLevel": {
+          "oneOf": [
+            {
+              "enum": [
+                "BoundedStaleness"
+              ]
+            },
+            { "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression" }
+          ]
+        },
+        "maxStalenessPrefix": {
+          "oneOf": [
+            {
+              "type": "integer",
+              "maximum": 1000,
+              "minimum": 10
+            },
+            { "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression" }
+          ]
+        },
+        "maxIntervalInSeconds": {
+          "oneOf": [
+            {
+              "type": "integer",
+              "maximum": 600,
+              "minimum": 5
+            },
+            { "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression" }
+          ]
+        }
+      },
+      "required": [
+        "defaultConsistencyLevel",
+        "maxIntervalInSeconds",
+        "maxStalenessPrefix"
+      ]
+    },
+    "consistencyOther": {
+      "type": "object",
+      "properties": {
+        "defaultConsistencyLevel": {
+          "oneOf": [
+            {
+              "enum": [
+                "Strong",
+                "Session",
+                "Eventual"
+              ]
+            },
+            { "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression" }
+          ]
+        }
+      },
+      "required": [
+        "defaultConsistencyLevel"
+      ]
+    }
```

Comment:
I would recommend adding "additionalProperties: false" here after the "required" property so that if a user switches from using "BoundedStaleness" over to one of the other consistencyLevels, they are reminded that "maxStalenessPrefix" and "maxIntervalInSeconds" don't apply to these other levels.